docs/Constraints.md

```diff
@@ -198,8 +199,47 @@ Only the currently active SDK (queryable via `dotnet --version`, changeable by t
 ```json
    "current-with-previews": {
        "web-assembly": {
-           "type": "sdk-varsion",
+           "type": "sdk-version",
            "args": "7.*.*-*"
        },
    }
-```
\ No newline at end of file
+```
+## Project capabilities
+Defines [project capabilities](https://github.com/microsoft/VSProjectSystem/blob/master/doc/overview/about_project_capabilities.md) that the template requires.
+Commonly used with item templates to define the certain projects it is applicable to. 
+`dotnet new` attempts to find the closest project file using following rules:
+- the project in current directory or `--output` directory (matching `*.*proj` extension)
+- if not found, the parent of above and so on
+
+In case of ambiguity, the path to the project can be specified using `--project` option.
+
+Once project is located, its project capabilities are evaluated. The project should be restored, otherwise evaluation fails.
+Only .NET [SDK-style projects](https://docs.microsoft.com/en-us/dotnet/core/project-sdk/overview) are supported.
+
+**Configuration:**
+
+- `type`: `project-capability`
+- `args`: (string) project capability [expression](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.shell.interop.vsprojectcapabilityexpressionmatcher) that should be satisfied by the project.
+
+**Supported in**:
+   - .NET SDK CLI (`dotnet new`)
+   - Visual Studio - TBD
```

Comment:
@phenning  is this feature ready in VS as below?

Reply:
Yes, this is in for 17.4 Preview 3